apps/passport-server/src/services/telegramService.ts

```diff
@@ -867,21 +837,20 @@ export class TelegramService {
     topicId: number
   ): Promise<string> {
     // Confirm that topicId exists and is anonymous
-    const topics = await fetchTelegramAnonTopicsByChatId(
+    const topic = await fetchTelegramTopic(
       this.context.dbPool,
-      telegramChatId
-    );
-    const topic = topics.find(
-      (t) => t.topic_id === topicId.toString() && t.is_anon_topic
+      telegramChatId,
+      topicId
```

Comment:
we need to check that topic is anon here?

Reply:
we do need to confirm topic is anon, and I wasn't checking. good catch!